### PR TITLE
[UX] Pageheader improvements

### LINKED ISSUE #1678: Declutter page header
There is a lot of information currently displayed at the top of each control header. E.g. `Namespace` and `Inheritance` information is tertiary information that is also listed in the WinUI documentation.

### PR DESCRIPTION
**Pull request by niels9001**
image

This PR moves the `Namespace` and `Inheritance` information in a flyout.

Closes #1678

### CODE REVIEW

**Comment by marcelwgn** [comment]
The button looks a bit off-center to me, it should probably be centered vertically to the main text run right?

File: WinUIGallery/ItemPage.xaml [hunk 1]
```diff
@@ -34,6 +34,32 @@
             <RowDefinition Height="Auto" />
             <RowDefinition Height="*" />
         </Grid.RowDefinitions>
+        <controls:PageHeader
```
Comment (marcelwgn) [inline WinUIGallery/ItemPage.xaml:37]: Related?

**Comment by Jay-o-Way** [comment]
A number of these are empty. Would be pretty to hide the pop-up or better the button in this case.

**Comment by niels9001** [comment]
> The button looks a bit off-center to me, it should probably be centered vertically to the main text run right?

Fixed it, it's now properly bottom aligned (text rendering and alignment is so much fun 😁)

**Comment by niels9001** [comment]
> A number of these are empty. Would be pretty to hide the pop-up or better the button in this case.

Great catch! That's being handled now, so the i button no longer appears when there is no Namespace and BaseClasses

File: WinUIGallery/Controls/PageHeader.xaml [hunk 2]
```diff
@@ -10,38 +10,90 @@
...
+                TextWrapping="NoWrap" />
+            <Button
+                x:Name="APIDetailsBtn"
+                Margin="0,0,0,1"
+                Padding="4"
+                VerticalAlignment="Bottom"
+                AutomationProperties.Name="API details"
+                Style="{StaticResource SubtleButtonStyle}"
+                ToolTipService.ToolTip="Details of the API like inheritance and namespace">
+                <FontIcon FontSize="14" Glyph="&#xE946;" />
```
Comment (Jay-o-Way) [inline WinUIGallery/Controls/PageHeader.xaml:50]:
```suggestion
                <FontIcon FontSize="16" Glyph="&#xE946;" />
```

File: WinUIGallery/Controls/PageHeader.xaml [hunk 3]
```diff
@@ -10,38 +10,90 @@
...
+                                    VerticalAlignment="Center"
+                                    FontFamily="Consolas"
+                                    IsTextSelectionEnabled="True"
+                                    Style="{StaticResource CaptionTextBlockStyle}"
+                                    Text="{x:Bind Item.ApiNamespace}" />
+                            </StackPanel>
+                            <MenuFlyoutSeparator Margin="-12,0,-12,0" />
+                            <StackPanel
+                                x:Name="InheritanceSourcePanel"
+                                Margin="0,0,0,4"
```
Comment (Jay-o-Way) [inline WinUIGallery/Controls/PageHeader.xaml:73]: It can also do without this

File: WinUIGallery/Controls/PageHeader.xaml (hunk 2 above)
Comment (niels9001) [reply WinUIGallery/Controls/PageHeader.xaml]: Great catch!

File: WinUIGallery/Controls/PageHeader.xaml [hunk 4]
```diff
@@ -10,38 +10,86 @@
...
+                                    Style="{StaticResource CaptionTextBlockStyle}"
+                                    Text="Namespace" />
+                                <TextBlock
+                                    VerticalAlignment="Center"
+                                    FontFamily="Consolas"
+                                    IsTextSelectionEnabled="True"
+                                    Style="{StaticResource CaptionTextBlockStyle}"
+                                    Text="{x:Bind Item.ApiNamespace}" />
+                            </StackPanel>
+                            <MenuFlyoutSeparator Margin="-12,0,-12,0" />
```
Comment (Jay-o-Way) [inline WinUIGallery/Controls/PageHeader.xaml:69]:
```suggestion
                            <MenuFlyoutSeparator Margin="-12" />
```
as used under the "Source" DropDown button

File: WinUIGallery/Controls/PageHeader.xaml [hunk 5]
```diff
@@ -10,38 +10,86 @@
...
+                TextTrimming="CharacterEllipsis"
+                TextWrapping="NoWrap" />
+            <Button
+                x:Name="APIDetailsBtn"
+                Margin="0,0,0,1"
+                Padding="4"
+                VerticalAlignment="Bottom"
+                AutomationProperties.Name="API details"
+                Style="{StaticResource SubtleButtonStyle}"
+                ToolTipService.ToolTip="API details like inheritance and namespace">
```
Comment (Jay-o-Way) [inline WinUIGallery/Controls/PageHeader.xaml:48]:
```suggestion
                ToolTipService.ToolTip="API namespace and inheritance">
```

**Comment by marcelwgn** [comment]
Is it just me or does the icon still look off center? 😅
Screenshot of header and offcenter button

**Comment by niels9001** [comment]
> Is it just me or does the icon still look off center? 😅 Screenshot of header and offcenter button

Pushed a small change. Should be bottom-aligned now!

image

**Review by marcelwgn — APPROVED** [review]

**Comment by Jay-o-Way** [comment]
@marcelwgn would somebody like to look at my pr? It's related...

**Pull request merged**